src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/ChangeFeed/SqlServerFhirResourceChangeDataStoreTests.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Health.Fhir.SqlServer.Features.ChangeFeed;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Configs;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.SqlServer.UnitTests.Features.ChangeFeed
+{
+    public class SqlServerFhirResourceChangeDataStoreTests
```

Comment:
question: is there a way that we can mock and have an end-to-end test? For example, mocking the results from the sqlfactory and sqlcommand, then validating the data processing and returning as well.

Reply:
The scenario is going to covered by an integration test which I am going to add after PR 2032.

Reply:
Integration tests are added to the PR.